app/src/main/java/com/conupods/OutdoorMaps/MapInitializer.java

```diff
@@ -7,28 +7,114 @@
 import android.widget.EditText;
 import android.widget.TextView;
 
+import com.conupods.IndoorMaps.IndoorOverlayHandlers.CCBuildingHandler;
+import com.conupods.IndoorMaps.IndoorOverlayHandlers.DefaultHandler;
+import com.conupods.IndoorMaps.IndoorOverlayHandlers.HallBuildingHandler;
+import com.conupods.IndoorMaps.IndoorOverlayHandlers.MBBuildingHandler;
+import com.conupods.IndoorMaps.IndoorOverlayHandlers.IndoorOverlayHandler;
+import com.conupods.IndoorMaps.IndoorBuildingOverlays;
+import com.conupods.IndoorMaps.IndoorOverlayHandlers.VLBuildingHandler;
 import com.conupods.R;
+import com.google.android.gms.maps.model.LatLngBounds;
 import com.google.android.gms.maps.GoogleMap;
 import com.google.android.gms.maps.model.Marker;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
 import static android.view.inputmethod.EditorInfo.IME_ACTION_SEARCH;
 
-public class MapInitializer {
+public class MapInitializer{
+
+
+    public static final double mZoomLevel = 18.3;
 
     private CameraController mCameraController;
     private GoogleMap mMap;
     private BuildingInfoWindow mBuildingInfoWindow;
+    private IndoorBuildingOverlays mIndoorBuildingOverlays;
+    private OutdoorBuildingOverlays mOutdoorBuildingOverlays;
     private static final String TAG = "MapInitializer";
 
-    public MapInitializer(CameraController CameraController, GoogleMap map, BuildingInfoWindow buildingInfoWindow) {
-        mCameraController = CameraController;
-        mMap = map;
+    private IndoorOverlayHandler chain1;
+    IndoorOverlayHandler chain2 = new MBBuildingHandler();
+    IndoorOverlayHandler chain3 = new VLBuildingHandler();
+    IndoorOverlayHandler chain4 = new CCBuildingHandler();
+    IndoorOverlayHandler defaultChain = new DefaultHandler();
```

Comment:
you dont need to create all these references in the class, u just need chain1, with a better name (Also member var notation).  you can create all those chains in the constructor